src/main/java/net/dv8tion/jda/internal/entities/AbstractChannelImpl.java

```diff
@@ -166,8 +164,26 @@ public boolean isSynced()
     {
         if (getParent() == null)
             return true; // Channels without a parent category are always considered synced. Also the case for categories.
-        
-        return Helpers.deepEqualsUnordered(getParent().getPermissionOverrides(), getPermissionOverrides());
+
+        Collection<PermissionOverride> thisCheck = getPermissionOverrides().stream().filter(
+                po -> po.getAllowedRaw() != 0 || po.getDeniedRaw() != 0
+        ).collect(Collectors.toList());
+
+        HashMap<Long, PermissionOverride> parentCheck = new HashMap<>();
+        getParent().getPermissionOverrides().stream().filter(
+                po -> po.getAllowedRaw() != 0 || po.getDeniedRaw() != 0
+        ).forEach(po -> parentCheck.put(po.getIdLong(), po));
+
+        for(PermissionOverride check : thisCheck)
+        {
+            PermissionOverride matching = parentCheck.getOrDefault(check.getIdLong(), null);
+            if(matching == null)
+                return false;
+
+            if(matching.getAllowedRaw() != check.getAllowedRaw() || matching.getDeniedRaw() != check.getDeniedRaw())
```

Comment:
I think the client only considers permissions that are valid for channels here. In some circumstances bots mess up permissions on channels and add stuff like `MANAGE_GUILD` or `ADMINISTRATOR` to channel overrides, we should try to filter that out.  A useful constant for this is [`Permission.ALL_CHANNEL_PERMISSIONS`](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/Permission.html#ALL_CHANNEL_PERMISSIONS)